includes/admin/class-sensei-home.php

```diff
@@ -0,0 +1,164 @@
+ *
+ * @since $$next-version$$
+ */
+final class Sensei_Home {
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Home constructor. Prevents other instances from being created outside `Sensei_Home::instance()`.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Initializes the class and adds all filters and actions related to Sensei Home.
+	 *
+	 * @since $$next-version$$
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_assets' ) );
+	}
+
+	/**
+	 * Enqueues admin scripts when needed on different screens.
+	 *
+	 * @since $$next-version$$
+	 * @access private
+	 */
+	public function enqueue_admin_assets() {
+		$screen = get_current_screen();
+
+		if ( 'course_page_sensei-home' === $screen->id ) {
+			Sensei()->assets->enqueue( 'sensei-home', 'home/index.js', [], true );
+			Sensei()->assets->enqueue( 'sensei-home-style', 'home/home.css', [ 'sensei-wp-components' ] );
+			Sensei()->assets->preload_data( [ '/sensei-internal/v1/sensei-extensions?type=plugin' ] );
+
+			$this->localize_script();
+		}
+	}
+
+	/**
+	 * Localize Home script.
+	 *
+	 * @since $$next-version$$
+	 */
+	private function localize_script() {
+		$data = array(
+			'connectUrl' => add_query_arg(
+				array(
+					'page'              => 'wc-addons',
+					'section'           => 'helper',
+					'wc-helper-connect' => 1,
+					'wc-helper-nonce'   => wp_create_nonce( 'connect' ),
+				),
+				admin_url( 'admin.php' )
+			),
+		);
+
+		if ( ! Sensei_Utils::is_woocommerce_installed() ) {
+			$data['installUrl'] = self_admin_url( 'plugin-install.php?s=woocommerce&tab=search&type=term&plugin_details=woocommerce' );
+		} elseif ( ! Sensei_Utils::is_woocommerce_active() ) {
+			$plugin_file         = 'woocommerce/woocommerce.php';
+			$data['activateUrl'] = wp_nonce_url(
+				add_query_arg(
+					array(
+						'action' => 'activate',
+						'plugin' => $plugin_file,
+					),
+					self_admin_url( 'plugins.php' )
+				),
+				'activate-plugin_' . $plugin_file
+			);
+		}
+
+		wp_localize_script(
+			'sensei-home',
+			'sensei_home',
+			$data
+		);
+	}
+
+	/**
+	 * Get updates count.
+	 *
+	 * @return int Updates count.
+	 */
+	private function get_has_update_count() {
+		$extensions = Sensei_Extensions::instance()->get_extensions( 'plugin' );
+
+		return count(
+			array_filter(
+				array_column( $extensions, 'has_update' )
+			)
+		);
+	}
+
+	/**
+	 * Adds the menu item for the Home page.
+	 *
+	 * @since  $$next-version$$
+	 *
+	 * @access private
+	 */
+	public function add_admin_menu_item() {
+		$updates_html = '';
+		$updates      = $this->get_has_update_count();
+
+		if ( $updates > 0 ) {
+			$updates_html = ' <span class="awaiting-mod">' . esc_html( $updates ) . '</span>';
+		}
+
+		add_submenu_page(
+			'edit.php?post_type=course',
+			__( 'Sensei LMS Home', 'sensei-lms' ),
+			__( 'Home', 'sensei-lms' ) . $updates_html,
+			'install_plugins',
```

Comment:
Hmmm. This is an interesting problem: do we want `Home` for everyone but just limit what we show based on individual user's permissions?

Reply:
I think based on p1663953676315749/1663948479.950919-slack-C8QJAM52Bwe probably should maintain the code as is, right? :)